vendor/lib/src/config.dart

```diff
@@ -0,0 +1,349 @@
+      );
+}
+
+/// Configuration of the contents for a vendored package.
+///
+/// The [VendoredSource] object contains the information necessary to create the
+/// contents of a vendored package folder.
+@sealed
+abstract class VendoredSource {
+  /// Name of package to vendored from `pub.dev`.
+  String get package;
+
+  /// Version of [package] to be vendored from `pub.dev`.
+  Version get version;
+
+  /// A map from _package name_ to _vendored package name_, to be rewritten in
+  /// this _vendored package_.
+  ///
+  /// **Example:** The following example will rewrite `package:foo/...` into
+  /// `lib/src/third_party/bar/...`.
+  /// ```
+  /// rewrites:
+  ///   foo: bar
+  /// ```
+  Map<String, String> get rewrites;
+
+  /// List of patterns for files/folders to include when vendoring.
+  ///
+  /// File patterns syntax hews closely to Bash glob-syntax, for details see:
+  /// https://pub.dev/packages/glob
+  ///
+  /// Filenames are relative to the root of the vendored package.
+  ///
+  /// If not specified this defaults to:
+  ///  * `pubspec.yaml`,
+  ///  * `README.md`,
+  ///  * `LICENSE`,
+  ///  * `CHANGELOG.md`,
+  ///  * `lib/**`, and,
+  ///  * `analysis_options.yaml`.
+  ///
+  /// See [_defaultIncludePatterns].
+  Set<String> get include;
+
+  @override
+  bool operator ==(Object other) {
+    if (other is! VendoredSource) {
+      return false;
+    }
+    if (rewrites.length != other.rewrites.length) {
+      return false;
+    }
+    for (final entry in rewrites.entries) {
+      if (other.rewrites[entry.key] != entry.value) {
+        return false;
+      }
+    }
+    if (!include.containsAll(other.include) ||
+        !other.include.containsAll(include)) {
+      return false;
+    }
+    return package == other.package && version == other.version;
+  }
+
+  @override
+  int get hashCode => Object.hash(
+        package,
+        version,
+        Object.hashAllUnordered(rewrites.entries.map((e) => Object.hash(
+              e.key,
+              e.value,
+            ))),
+        Object.hashAllUnordered(include),
+      );
+}
+
+@JsonSerializable(
+  anyMap: true,
+  checked: true,
+  disallowUnrecognizedKeys: true,
+  includeIfNull: false,
+)
+class _VendorState extends VendorState {
+  @override
+  @JsonKey(
+    required: true,
+    fromJson: _versionFromJson,
+    toJson: _versionToJson,
+  )
+  final Version version;
+
+  @override
+  @JsonKey(required: true)
+  final _VendorConfig config;
+
+  _VendorState(this.version, this.config);
+
+  static _VendorState fromJson(Map json) => _$VendorStateFromJson(json);
+
+  @override
+  String toYaml() {
+    // Encode / decode JSON to ensure that _VendorConfig.toJson() is called
+    final raw = json.decode(json.encode(_$VendorStateToJson(this)));
+    final e = YamlEditor('');
+    e.update([], wrapAsYamlNode(raw, collectionStyle: CollectionStyle.BLOCK));
+    return [
+      '# DO NOT EDIT: This file is generated by package:vendor version $packageVersion',
```

Comment:
We should totally have a way of adding comments in yaml-edit!!

Reply:
Yes